src/Nethermind/Nethermind.Consensus/Producers/TxPoolTxSource.cs

```diff
@@ -77,51 +127,32 @@ public IEnumerable<Transaction> GetTransactions(BlockHeader parent, long gasLimi
                 bool success = _txFilterPipeline.Execute(tx, parent);
                 if (!success) continue;
 
-                if (tx.SupportsBlobs)
+                if (_logger.IsTrace) _logger.Trace($"Selected {tx.ToShortString()} to be potentially included in block.");
+
+                if (selectedBlobTxs?.Count > 0)
                 {
-                    if (blobGasPrice.IsZero)
+                    foreach (Transaction blobTx in new List<Transaction>(selectedBlobTxs))
```

Comment:
It is because we need to mix blob and non-blob transactions and return in correct order. So firstly we are picking best blob transactions (up to 6 blobs cumulatively, so max 6 txs), we can't just yield return it because we don't know if these txs are better than non-blob ones. So we have them on the list. Then we are iterating non-blob txs. Before returning  every non-blob tx, we are iterating blob txs and yield returning if is better than non-blob tx, and removing blob tx from the list. This way we have mixed blob and non-blob txs yield returned in correct order. At the end we are returning remaining blob transactions, when there is no more non-blob ones.

In this place we need to allocate another list, because we will remove txs from initial one and we can't iterate through collection and remove from it during iteration. But unfortunately it will allocate on every iteration of non-blob tx, so I will refactor it to use already existing list and create another one with items to remove from main one